khanh19934/backend-coding-test
# Refactor code and restructure project

## What problem does this PR solve

- Refactor and restructure project
- Separate logic and test independent

## How does this PR solve it

- Separate logic for easy maintenance and bug fixing, easy for reading
- Write test independent to up more coverage

## How can reviewers test the changes

- Check the code change
- Run test to check it should pass test case

## What Documentation/References will help understand the problem and solution

(Please be sure to include links to any libraries, conversations, post, etc...)

## Any limitation or related work yet to be done

- Still can not mock the db call to testing, if can we can push the coverage to 100%, now is only > 90%
![Screen Shot 2021-08-21 at 13 47 41](https://user-images.githubusercontent.com/18601105/130313336-975d8ad7-3abc-412d-a318-14588f53158f.png)
